### PR TITLE
added concat_img() in nilearn.image

### PR DESCRIPTION
**Pull request by banilo**
tries to address #183

### CODE REVIEW

**Comment by lesteve** [comment]
An additional thing is that the promise of this PR is not fulfilled: you have not added concat_img to nilearn.image ...

I think the idea was just to do something along the lines of:
`from _utils.niimg_conversions import concat_niimg as concat_img`
in `nilearn/image/__init__.py`. A test using nilearn.image.concat_img would be good.

**Comment by banilo** [comment]
All aspects should have been addressed now.

**Comment by AlexandreAbraham** [comment]
Is there a usecase where we want to concat 3D and 4D images ? I would expect to concat only 3D or 4D images, mixing 3D and 4D images should raise an error, or at least a warning.

File: nilearn/image/tests/test_image.py [hunk 1]
```diff
@@ -271,3 +272,13 @@ def test_swap_img_hemispheres():
         data)
 
 
+def test_concat_imgs():
+    # simply smoke-test the import from _utils.concat_niimgs
```
Comment (lesteve) [inline nilearn/image/tests/test_image.py]: You could do something like this that would test that they are actually the exact same function

```python
# in the imports
from ..._utils import niimg_conversions

# in the tests
def test_concat_imgs():
    image.concat_imgs is niimg_conversions.concat_niimgs
```

File: nilearn/_utils/niimg_conversions.py [hunk 2]
```diff
@@ -222,42 +248,61 @@ def concat_niimgs(niimgs, dtype=np.float32, accept_4d=False):
...
-            if (isinstance(iter_niimg, basestring)):
-                i_error = "image " + iter_niimg
-            else:
-                i_error = "image #" + str(index)
-            raise ValueError("Shape of %s is different from first image shape."
-                             % i_error)
-        data[..., index:index + size] = _to_4d(this_data)
-    return nibabel.Nifti1Image(data, affine)
+
+        # parse current 3D or 4D nifti or path to such
```
Comment (lesteve) [inline nilearn/_utils/niimg_conversions.py]: I don't understand what you mean by "parse" here and also a few times in the lines below. Also I am not sure what useful piece of information this comments bring.

File: nilearn/image/tests/test_image.py (hunk 1 above)
Comment (lesteve) [inline nilearn/image/tests/test_image.py]: Sorry there is a missing assert_true in the previous comment:
 
```python
# in the tests
def test_concat_imgs():
    assert_true(image.concat_imgs is niimg_conversions.concat_niimgs)
```

File: nilearn/image/tests/test_image.py [hunk 3]
```diff
@@ -271,3 +272,15 @@ def test_swap_img_hemispheres():
         data)
 
 
+def test_concat_imgs():
+    assert_true(concat_imgs is niimg_conversions.concat_niimgs)
+    
+    # simply smoke-test the import from _utils.concat_niimgs
```
Comment (lesteve) [inline nilearn/image/tests/test_image.py]: I guess you don't need this test any more, i.e. all the testing is done in the _utils.niimg_conversions test

**Comment by lesteve** [comment]
> Is there a usecase where we want to concat 3D and 4D images ? I would expect to concat only 3D or 4D images, mixing 3D and 4D images should raise an error, or at least a warning.

Not sure but I don't think @banilo changed the behaviour of concat_niimgs. Do open an issue if you feel strongly about it.

**Comment by banilo** [comment]
As @lesteve said, this PR does not relate to the behavior you are talking about. This aspect remains unchanged.

**Comment by lesteve** [comment]
OK merging this one, good work!

**Comment by banilo** [comment]
:beer:

**Pull request merged**